### PR TITLE
resthopper refactor + minor enhancements

### PR DESCRIPTION
**Pull request by cdriesler**
Refactors existing resthopper functionality to:

- remove double-iteration through all objects in definition
- remove duplication within each geometry type's switch-case block
- shorten and generalize input/output param parsing logic (mostly happens in `Schema.cs`)

Main goal was to make future changes to the input/output schema easier to implement and maintain.

Adds the following for clarity and error handling:

- comments comments comments
- `400` error code responses with explanations (taking some of the burden off the current `PayAttentionException`
- a `ResthopperParam` class to compliment the `ResthopperObject` class

Big fan of your implementation here @enmerk4r ! Figured I could kill two birds with one stone and contribute a little bit while I studied it.

### CODE REVIEW

File: src/compute.geometry/IO/Schema.cs [hunk 1]
```diff
@@ -61,4 +67,215 @@ public ResthopperObject(object obj)
...
+                    var data = GetGeometry(restobj.Data, TypeCode);
+                    dynamic ghObj = Activator.CreateInstance(assembly.GetType($"Grasshopper.Kernel.Types.{TypeCodeToValueType[TypeCode]}"), data);
+                    typedParam.AddVolatileData(path, i, ghObj);
+                }
+            }
+
+            return true;
+        }
+
+        private dynamic GetGeometry(string json, GHTypeCodes type)
```
Comment (cdriesler) [inline src/compute.geometry/IO/Schema.cs]: haven't figured out dynamic deserialization just yet...but I think this is better than the repeating code from before

File: src/compute.geometry/IO/Schema.cs [hunk 2]
```diff
@@ -61,4 +67,215 @@ public ResthopperObject(object obj)
...
+                Name = labelData[2];
+            }
+
+            return true;
+        }
+    }
+
+    public class ResthopperInput : ResthopperParam
+    {
+        public static Dictionary<GHTypeCodes, string> TypeCodeToParamType { get; } = new Dictionary<GHTypeCodes, string>()
```
Comment (cdriesler) [inline src/compute.geometry/IO/Schema.cs]: if we got picky with naming conventions, these dictionaries could get cut out and we could just make minor alterations to the `GHTypeCode` name

**Comment by sbaer** [comment]
There are a lot of changes here.  I'm sorry, but I can't figure out how all of these changes work to a point where I feel comfortable merging this PR

**Comment by cdriesler** [comment]
> 
> 
> There are a lot of changes here. I'm sorry, but I can't figure out how all of these changes work to a point where I feel comfortable merging this PR

Totally understood. First time trying to contribute like this, will break it up into much smaller pieces and try again.

**Pull request closed without merging**